### PR TITLE
typecore: comment the backtracking logic in type_label_exp

### PR DESCRIPTION
**Pull request by gasche**
This is a refactoring follow-up of #11536, a second try after #11900 failed to gather support.

#11900 tried to clarify the code structure itself, but was found unconvincing.
The present PR merely comments the existing code.

(cc @goldfirere)

### CODE REVIEW

File: typing/typecore.ml [hunk 1]
```diff
@@ -4352,18 +4357,22 @@ and type_label_exp create env loc ty_expected
       let arg = type_argument env sarg ty_arg (instance ty_arg) in
       (vars, ty_arg, snap, arg)
     end
+    (* Note: there is no generalization logic here as could be expected,
+       because it is part of the backtracking lgoic below. *)
```
Comment (t6s) [inline typing/typecore.ml]: typo: lgoic -> logic

Comment (gasche) [reply typing/typecore.ml]: Thanks!

**Review by goldfirere — APPROVED** [review]

**Comment by gasche** [comment]
Thanks!

(This could probably be approved on behalf of @goldfirere and merged, but I cannot do it myself.)

**Review by Octachron — APPROVED** [review]
The comments look fine to me too.

**Pull request merged**